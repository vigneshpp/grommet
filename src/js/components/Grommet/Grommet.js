import React, { forwardRef, useEffect, useMemo, useState } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ContainerTargetContext,
  ResponsiveContext,
  ThemeContext,
} from '../../contexts';

import {
  deepMerge,
  backgroundIsDark,
  getBreakpoint,
  getDeviceBreakpoint,
  normalizeColor,
  useForwardedRef,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';
import { RootsContext } from '../../contexts/RootsContext';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

const deviceResponsive = (userAgent, theme) => {
  // log('--deviceResponsive', userAgent, theme);
  /*
   * Regexes provided for mobile and tablet detection are meant to replace
   * a full-featured specific library due to contributing a considerable size
   * into the bundle.
   *
   * User agents found https://deviceatlas.com/blog/list-of-user-agent-strings
   */
  if (userAgent) {
    if (/(tablet|ipad|playbook|silk)|(android(?!.*mobile))/i.test(userAgent)) {
      return getDeviceBreakpoint('tablet', theme);
    }
    if (/Mobile|iPhone|Android/.test(userAgent)) {
      return getDeviceBreakpoint('phone', theme);
    }
    return getDeviceBreakpoint('computer', theme);
  }
  return undefined;
};

const Grommet = forwardRef((props, ref) => {
  const {
    children,
    full,
    containerTarget = typeof document === 'object' ? document.body : undefined,
    theme: themeProp,
    ...rest
  } = props;

  const { background, dir, themeMode, userAgent } = props;

  const [stateResponsive, setResponsive] = useState();

  const theme = useMemo(() => {
    const nextTheme = deepMerge(baseTheme, themeProp || {});

    // if user provides specific menu alignment, we don't want
    // the defaults to be included at all (can cause issues with controlMirror)
    // override merged value with themeProp value
    if (
      themeProp &&
      themeProp.menu &&
      themeProp.menu.drop &&
      themeProp.menu.drop.align
    ) {
      delete nextTheme.menu.drop.align;
      nextTheme.menu.drop.align = themeProp.menu.drop.align;
    }
    const {
      colors: { background: themeBackground },
    } = nextTheme.global;

    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';
    const color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;

    if (dir) {
      nextTheme.dir = dir;
    }

    return nextTheme;
  }, [background, dir, themeMode, themeProp]);

  useEffect(() => {
    const onResize = () => {
      setResponsive(getBreakpoint(document.body.clientWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);

  const responsive =
    stateResponsive ||
    deviceResponsive(userAgent, theme) ||
    theme.global.deviceBreakpoints.tablet;

  const grommetRef = useForwardedRef(ref);

  return (
    <ThemeContext.Provider value={theme}>
      <ResponsiveContext.Provider value={responsive}>
        <RootsContext.Provider value={[grommetRef.current]}>
          <ContainerTargetContext.Provider value={containerTarget}>
            <StyledGrommet full={full} {...rest} ref={grommetRef}>
              {children}
            </StyledGrommet>
            {full && <FullGlobalStyle />}
          </ContainerTargetContext.Provider>
        </RootsContext.Provider>
      </ResponsiveContext.Provider>
    </ThemeContext.Provider>
  );
});

Grommet.displayName = 'Grommet';

let GrommetDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  GrommetDoc = require('./doc').doc(Grommet);
}
const GrommetWrapper = GrommetDoc || Grommet;

export { GrommetWrapper as Grommet };
