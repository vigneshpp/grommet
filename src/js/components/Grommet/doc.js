import { describe, PropTypes } from 'react-desc';

import { backgroundDoc } from '../../utils/prop-types';
import { getAvailableAtBadge } from '../../utils/mixins';

export const doc = Grommet => {
  const DocumentedGrommet = describe(Grommet)
    .availableAt(getAvailableAtBadge('Grommet', 'Utilities'))
    .description('The top level Grommet container.')
    .usage(
      `import { Grommet } from 'grommet';
<Grommet>...</Grommet>`,
    )
    .intrinsicElement('div');

  DocumentedGrommet.propTypes = {
    background: backgroundDoc,
    dir: PropTypes.oneOf(['rtl']).description(
      'Layout direction for right to left contexts',
    ),
    full: PropTypes.bool
      .description('Whether to take the whole viewport.')
      .defaultValue(false),
    plain: PropTypes.bool
      .description(
        `Whether or not Grommet should apply a global font-family, font-size,
        and line-height.`,
      )
      .defaultValue(false),
    cssVars: PropTypes.bool
      .description('Whether to expose the css variables.')
      .defaultValue(false),
    theme: PropTypes.object.description(
      'Custom styles for Grommet app component.',
    ),
    themeMode: PropTypes.oneOf(['dark', 'light']).description(
      `Dark vs. light theme variation. Default is unspecified and left to
      theme.`,
    ),
    userAgent: PropTypes.string.description(
      `User agent used to detect the device width for setting the initial
      breakpoint.`,
    ),
    containerTarget: PropTypes.object.description(
      `The node where Drop and Layer containers are inserted. Defaults to
      document.body which is almost always the right choice. This is used
      for less common cases like rendering within an internal node (e.g.
      shadow root).`,
    ),
  };

  return DocumentedGrommet;
};

export const themeDoc = {
  'grommet.extend': {
    description: 'Any additional style for Grommet.',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
};
