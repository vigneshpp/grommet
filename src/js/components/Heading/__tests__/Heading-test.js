import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Grommet } from '../../Grommet';
import { Heading } from '..';

test('Heading renders', () => {
  const component = renderer.create(
    <Grommet>
      <Heading />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Heading accepts ref', () => {
  const ref = React.createRef();
  const component = renderer.create(
    <Grommet>
      <Heading ref={ref} />
    </Grommet>,
    { createNodeMock: el => el },
  );
  expect(ref.current).not.toBeNull();
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Heading level renders', () => {
  const component = renderer.create(
    <Grommet>
      <Heading level={1} />
      <Heading level={2} />
      <Heading level={3} />
      <Heading level={4} />
      <Heading level="1" />
      <Heading level="2" />
      <Heading level="3" />
      <Heading level="4" />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Heading size renders', () => {
  const component = renderer.create(
    <Grommet>
      <Heading level={1} size="small" />
      <Heading level={1} size="medium" />
      <Heading level={1} size="large" />
      <Heading level={1} size="xlarge" />
      <Heading level={2} size="small" />
      <Heading level={2} size="medium" />
      <Heading level={2} size="large" />
      <Heading level={2} size="xlarge" />
      <Heading level={3} size="small" />
      <Heading level={3} size="medium" />
      <Heading level={3} size="large" />
      <Heading level={3} size="xlarge" />
      <Heading level={4} size="small" />
      <Heading level={4} size="medium" />
      <Heading level={4} size="large" />
      <Heading level={4} size="xlarge" />
      <Heading level={1} size="77px" />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Heading textAlign renders', () => {
  const component = renderer.create(
    <Grommet>
      <Heading textAlign="start" />
      <Heading textAlign="center" />
      <Heading textAlign="end" />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Heading margin renders', () => {
  const component = renderer.create(
    <Grommet>
      <Heading margin="small" />
      <Heading margin="medium" />
      <Heading margin="large" />
      <Heading margin="none" />
      <Heading margin={{ bottom: 'small' }} />
      <Heading margin={{ top: 'small' }} />
      <Heading margin={{ bottom: 'none' }} />
      <Heading margin={{ top: 'none' }} />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Heading color renders', () => {
  const component = renderer.create(
    <Grommet>
      <Heading color="brand" />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

const LONG = 'a b c d e f g h i j k l m n o p q r s t u v w x y z';

test('Heading truncate renders', () => {
  const component = renderer.create(
    <Grommet>
      <Heading truncate={false}>{LONG}</Heading>
      <Heading truncate>{LONG}</Heading>
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('responsive renders', () => {
  const component = renderer.create(
    <Grommet>
      <Heading responsive />
      <Heading responsive={false} />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Theme based font family renders', () => {
  const customTheme = {
    heading: {
      font: {
        family: 'Fira Sans',
      },
      level: {
        1: {
          font: {
            family: 'Arial',
          },
        },
        2: {
          font: {
            family: 'Roboto',
          },
        },
        3: {
          font: {
            family: 'Ubuntu',
          },
        },
      },
    },
  };
  const component = renderer.create(
    <Grommet theme={customTheme}>
      <Heading level={1} />
      <Heading level={2} />
      <Heading level={3} />
      <Heading level={4} />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Theme based font weight renders', () => {
  const customTheme = {
    heading: {
      weight: 600,
      level: {
        1: {
          font: {
            weight: '700',
          },
        },
        2: {
          font: {
            weight: '400',
          },
        },
        3: {
          font: {
            weight: '200',
          },
        },
      },
    },
  };
  const component = renderer.create(
    <Grommet theme={customTheme}>
      <Heading level={1} />
      <Heading level={2} />
      <Heading level={3} />
      <Heading level={4} />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Theme color renders', () => {
  const customTheme = {
    heading: {
      color: 'text-strong',
    },
  };
  const component = renderer.create(
    <Grommet theme={customTheme}>
      <Heading level={1} />
      <Heading level={2} />
      <Heading level={3} />
      <Heading level={4} />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Throws a warning when heading.level is undefined in the theme.', () => {
  global.console = {
    warn: jest.fn(),
  };

  const customTheme = {
    heading: {
      level: {
        '6': undefined,
      },
    },
  };

  renderer.create(
    <Grommet theme={customTheme}>
      <Heading level={6} />
    </Grommet>,
  );

  const consoleMsg = 'Heading level 6 is not defined in your theme.';
  expect(global.console.warn).toHaveBeenCalledWith(consoleMsg);
});

test('Heading fill renders', () => {
  const component = renderer.create(
    <Grommet>
      <Heading fill />
    </Grommet>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
