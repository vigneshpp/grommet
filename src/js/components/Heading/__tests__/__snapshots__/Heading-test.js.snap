// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading accepts ref 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
</div>
`;

exports[`Heading color renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  color: #7D4CDB;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
</div>
`;

exports[`Heading fill renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: none;
  font-weight: 600;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: none;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
</div>
`;

exports[`Heading level renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c2 {
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
}

.c3 {
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
  font-weight: 600;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    font-size: 16px;
    line-height: 22px;
    max-width: 384px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h2
    className="c2"
  />
  <h3
    className="c3"
  />
  <h4
    className="c4"
  />
  <h1
    className="c1"
  />
  <h2
    className="c2"
  />
  <h3
    className="c3"
  />
  <h4
    className="c4"
  />
</div>
`;

exports[`Heading margin renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  margin: 12px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c2 {
  margin: 24px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c3 {
  margin: 48px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c4 {
  margin: 0px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c5 {
  margin-bottom: 12px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c6 {
  margin-top: 12px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c7 {
  margin-bottom: 0px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c8 {
  margin-top: 0px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    margin-top: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    margin-bottom: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    margin-top: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h1
    className="c2"
  />
  <h1
    className="c3"
  />
  <h1
    className="c4"
  />
  <h1
    className="c5"
  />
  <h1
    className="c6"
  />
  <h1
    className="c7"
  />
  <h1
    className="c8"
  />
</div>
`;

exports[`Heading renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
</div>
`;

exports[`Heading size renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
}

.c2 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c3 {
  font-size: 82px;
  line-height: 88px;
  max-width: 1968px;
  font-weight: 600;
}

.c4 {
  font-size: 114px;
  line-height: 120px;
  max-width: 2736px;
  font-weight: 600;
}

.c5 {
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
}

.c6 {
  font-size: 66px;
  line-height: 72px;
  max-width: 1584px;
  font-weight: 600;
}

.c7 {
  font-size: 22px;
  line-height: 28px;
  max-width: 528px;
  font-weight: 600;
}

.c8 {
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
}

.c9 {
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
}

.c10 {
  font-size: 42px;
  line-height: 48px;
  max-width: 1008px;
  font-weight: 600;
}

.c11 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
  font-weight: 600;
}

.c12 {
  font-size: 77px;
  line-height: normal;
  max-width: 1200px;
  font-weight: 600;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    font-size: 50px;
    line-height: 56px;
    max-width: 1200px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    font-size: 66px;
    line-height: 72px;
    max-width: 1584px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    font-size: 22px;
    line-height: 28px;
    max-width: 528px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    font-size: 42px;
    line-height: 48px;
    max-width: 1008px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width:768px) {
  .c10 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    font-size: 16px;
    line-height: 22px;
    max-width: 384px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
    size="small"
  />
  <h1
    className="c2"
    size="medium"
  />
  <h1
    className="c3"
    size="large"
  />
  <h1
    className="c4"
    size="xlarge"
  />
  <h2
    className="c5"
    size="small"
  />
  <h2
    className="c1"
    size="medium"
  />
  <h2
    className="c2"
    size="large"
  />
  <h2
    className="c6"
    size="xlarge"
  />
  <h3
    className="c7"
    size="small"
  />
  <h3
    className="c8"
    size="medium"
  />
  <h3
    className="c9"
    size="large"
  />
  <h3
    className="c10"
    size="xlarge"
  />
  <h4
    className="c11"
    size="small"
  />
  <h4
    className="c11"
    size="medium"
  />
  <h4
    className="c11"
    size="large"
  />
  <h4
    className="c11"
    size="xlarge"
  />
  <h1
    className="c12"
    size="77px"
  />
</div>
`;

exports[`Heading textAlign renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  text-align: left;
}

.c2 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  text-align: center;
}

.c3 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  text-align: right;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h1
    className="c2"
  />
  <h1
    className="c3"
  />
</div>
`;

exports[`Heading truncate renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c2 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </h1>
  <h1
    className="c2"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </h1>
</div>
`;

exports[`Theme based font family renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-family: Arial;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c2 {
  font-family: Roboto;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
}

.c3 {
  font-family: Ubuntu;
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
}

.c4 {
  font-family: Fira Sans;
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
  font-weight: 600;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    font-size: 16px;
    line-height: 22px;
    max-width: 384px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h2
    className="c2"
  />
  <h3
    className="c3"
  />
  <h4
    className="c4"
  />
</div>
`;

exports[`Theme based font weight renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 700;
}

.c2 {
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 400;
}

.c3 {
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 200;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
  font-weight: 600;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    font-size: 16px;
    line-height: 22px;
    max-width: 384px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h2
    className="c2"
  />
  <h3
    className="c3"
  />
  <h4
    className="c4"
  />
</div>
`;

exports[`Theme color renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  color: #000000;
}

.c2 {
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  color: #000000;
}

.c3 {
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
  color: #000000;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
  font-weight: 600;
  color: #000000;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    font-size: 16px;
    line-height: 22px;
    max-width: 384px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h2
    className="c2"
  />
  <h3
    className="c3"
  />
  <h4
    className="c4"
  />
</div>
`;

exports[`responsive renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c2 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h1
    className="c2"
  />
</div>
`;
